csync/src/vio/csync_vio_local.c

```diff
@@ -44,23 +44,57 @@
  */
 
 typedef struct dhandle_s {
-  _TDIR *dh;
+#if defined _WIN32
+  WIN32_FIND_DATA ffd;
+  HANDLE hFind;
+  int firstFind;
+#else
+  DIR *dh;
+#endif
   char *path;
 } dhandle_t;
 
 csync_vio_handle_t *csync_vio_local_opendir(const char *name) {
   dhandle_t *handle = NULL;
-  mbchar_t *dirname = c_utf8_to_locale(name);
+  mbchar_t *dirname = NULL;
 
   handle = c_malloc(sizeof(dhandle_t));
 
+#ifdef _WIN32
+  // the file wildcard has to be attached
+  int len_name = strlen(name);
+  if( len_name ) {
+      char *h = NULL;
+
+      // alloc an enough large buffer to take the name + '/*' + the closing zero.
+      h = c_malloc(len_name+3);
+      strncpy( h, name, 1+len_name);
+      strncat(h, "/*",3);
```

Comment:
the 3 should actually be 2,  but it does not matter since it is a raw litteral.
I personaly would disagree with @ckamm here and say that strcpy and strcat were just fine.